src/main/java/org/wso2/extension/siddhi/window/uniquelengthbatch/UniqueLengthBatchWindowProcessor.java

```diff
@@ -0,0 +1,283 @@
+    private boolean isFirstUniqueEnabled = false;
+
+    /**
+     * The init method of the WindowProcessor, this method will be called before other methods.
+     *
+     * @param attributeExpressionExecutors the executors of each function parameters
+     * @param executionPlanContext         the context of the execution plan
+     */
+    @Override
+    protected void init(ExpressionExecutor[] attributeExpressionExecutors, ExecutionPlanContext executionPlanContext) {
+        this.executionPlanContext = executionPlanContext;
+        this.eventsToBeExpired = new ComplexEventChunk<>(false);
+        if (attributeExpressionExecutors.length == 2) {
+            this.uniqueKey = (VariableExpressionExecutor) attributeExpressionExecutors[0];
+            if (attributeExpressionExecutors[1] instanceof ConstantExpressionExecutor) {
+                if (attributeExpressionExecutors[1].getReturnType() == Attribute.Type.INT) {
+                    this.windowLength = (Integer)
+                            (((ConstantExpressionExecutor) attributeExpressionExecutors[1]).getValue());
+                } else {
+                    throw new ExecutionPlanValidationException("Length should be an INT");
+                }
+            } else {
+                throw new ExecutionPlanValidationException("Unique Length Batch window should have constant " +
+                        "for length parameter but found a dynamic attribute "
+                        + attributeExpressionExecutors[1].getClass().getCanonicalName());
+            }
+        } else if (attributeExpressionExecutors.length == 3) {
+            this.uniqueKey = (VariableExpressionExecutor) attributeExpressionExecutors[0];
+            if (attributeExpressionExecutors[1] instanceof ConstantExpressionExecutor) {
+                if (attributeExpressionExecutors[1].getReturnType() == Attribute.Type.INT) {
+                    this.windowLength = (Integer)
+                            (((ConstantExpressionExecutor) attributeExpressionExecutors[1]).getValue());
+                } else {
+                    throw new ExecutionPlanValidationException("Length should be an INT");
+                }
+            } else {
+                throw new ExecutionPlanValidationException("Unique Length Batch window should have constant " +
+                        "for length parameter but found a dynamic attribute "
+                        + attributeExpressionExecutors[1].getClass().getCanonicalName());
+            }
+            if (attributeExpressionExecutors[2] instanceof ConstantExpressionExecutor) {
+                if (attributeExpressionExecutors[2].getReturnType() == Attribute.Type.BOOL) {
+                    this.isFirstUniqueEnabled = (boolean) (((ConstantExpressionExecutor)
+                            attributeExpressionExecutors[2]).getValue());
+                } else {
+                    throw new ExecutionPlanValidationException("Third Value should be boolean value");
```

Comment:
This should be "Third parameter". Let's try to improve this error message as well, as explained above.